### ISSUE TITLE
Location is wrong and location marker isn't pulsing in "Animate the user icon" activity

### ISSUE THREAD

**Issue by langsmith** [mapbox/mapbox-android-demo#212]
Purple marker shows up but isn't pulsing like it should be. Location seems to be off too (i.e. purple marker is in the wrong location). To double-check, I uninstalled the app, re-installed, accepted location permission, and it still put me off the west coast of Africa.

![screenshot](https://cloud.githubusercontent.com/assets/4394910/22830139/672a74e0-ef73-11e6-8582-47b280ef9de5.png)

**Comment by tobrun** [comment]
I looked into the MarkerView animation part of this issue, 
if you use the callback for the markerview you are able to animate it:

![ezgif com-video-to-gif 15](https://cloud.githubusercontent.com/assets/2151639/23209349/3fae01c2-f8f9-11e6-9883-3a1b9737aa90.gif)


Code for that is:

```java
package com.mapbox.mapboxsdk.testapp.activity.maplayout;

import android.animation.AnimatorSet;
import android.animation.ObjectAnimator;
import android.animation.ValueAnimator;
import android.content.Context;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.mapbox.mapboxsdk.annotations.MarkerView;
import com.mapbox.mapboxsdk.annotations.MarkerViewManager;
import com.mapbox.mapboxsdk.geometry.LatLng;
import com.mapbox.mapboxsdk.maps.MapView;
import com.mapbox.mapboxsdk.maps.MapboxMap;
import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
import com.mapbox.mapboxsdk.testapp.R;
import com.mapbox.mapboxsdk.testapp.model.annotations.PulseMarkerView;
import com.mapbox.mapboxsdk.testapp.model.annotations.PulseMarkerViewOptions;

public class SimpleMapActivity extends AppCompatActivity {

  private MapView mapView;


  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_map_simple);

    mapView = (MapView) findViewById(R.id.mapView);
    mapView.onCreate(savedInstanceState);

    mapView.getMapAsync(new OnMapReadyCallback() {
      @Override
      public void onMapReady(final MapboxMap mapboxMap) {
        mapboxMap.getMarkerViewManager().addMarkerViewAdapter(new PulseMarkerViewAdapter(SimpleMapActivity.this));
        mapboxMap.addMarker(new PulseMarkerViewOptions().position(new LatLng()), new MarkerViewManager.OnMarkerViewAddedListener() {
          @Override
          public void onViewAdded(@NonNull MarkerView markerView) {
            View view = mapboxMap.getMarkerViewManager().getView(markerView);
            if (view != null) {
              View backgroundView = view.findViewById(R.id.background_imageview);

              ValueAnimator scaleCircleX = ObjectAnimator.ofFloat(backgroundView, "scaleX", 1.8f);
              ValueAnimator scaleCircleY = ObjectAnimator.ofFloat(backgroundView, "scaleY", 1.8f);
              ObjectAnimator fadeOut = ObjectAnimator.ofFloat(backgroundView, "alpha", 1f, 0f);

              scaleCircleX.setRepeatCount(ValueAnimator.INFINITE);
              scaleCircleY.setRepeatCount(ValueAnimator.INFINITE);
              fadeOut.setRepeatCount(ObjectAnimator.INFINITE);

              AnimatorSet animatorSet = new AnimatorSet();
              animatorSet.play(scaleCircleX).with(scaleCircleY).with(fadeOut);
              animatorSet.setDuration(1000);
              animatorSet.start();
            }
          }
        });
      }
    });
  }


  // Custom marker view used for pulsing the background view of marker.
  private static class PulseMarkerViewAdapter extends MapboxMap.MarkerViewAdapter<PulseMarkerView> {

    private LayoutInflater inflater;

    public PulseMarkerViewAdapter(@NonNull Context context) {
      super(context);
      this.inflater = LayoutInflater.from(context);
    }

    @Nullable
    @Override
    public View getView(@NonNull PulseMarkerView marker, @Nullable View convertView, @NonNull ViewGroup parent) {
      ViewHolder viewHolder;
      if (convertView == null) {
        viewHolder = new ViewHolder();
        convertView = inflater.inflate(R.layout.view_pulse_marker, parent, false);
        viewHolder.foregroundImageView = (ImageView) convertView.findViewById(R.id.foreground_imageView);
        viewHolder.backgroundImageView = (ImageView) convertView.findViewById(R.id.background_imageview);
        convertView.setTag(viewHolder);
      }
      return convertView;
    }

    private static class ViewHolder {
      ImageView foregroundImageView;
      ImageView backgroundImageView;
    }
  }

  @Override
  protected void onStart() {
    super.onStart();
    mapView.onStart();
  }

  @Override
  protected void onResume() {
    super.onResume();
    mapView.onResume();
  }

  @Override
  protected void onPause() {
    super.onPause();
    mapView.onPause();
  }

  @Override
  protected void onStop() {
    super.onStop();
    mapView.onStop();
  }

  @Override
  public void onLowMemory() {
    super.onLowMemory();
    mapView.onLowMemory();
  }

  @Override
  protected void onDestroy() {
    super.onDestroy();
    mapView.onDestroy();
  }

  @Override
  protected void onSaveInstanceState(Bundle outState) {
    super.onSaveInstanceState(outState);
    mapView.onSaveInstanceState(outState);
  }
}

```

**Comment by zugaldia** [comment]
@langsmith Is this the same issue as the one described in https://github.com/mapbox/mapbox-android-demo/issues/241?

**Comment by langsmith** [comment]
Kinda' @zugaldia. I'm still considering them as two separate issues.
Tobrun's code above makes the icon pulse in the correct user location....when the app is installed is installed and the example's opened for the first time.

When I kill the app, re-open the example, the marker **_doesn't_** pulse anymore but it's still in the correct user location. I'm still trying to figure out why that might be...

**Comment by cammace** [comment]
Sounds like we start the animation when permissions are granted but aren't starting when the user just starts the activity after the permissions already been granted. Haven't checked the code however.

**Comment by langsmith** [comment]
Yea, that's what I figured too @cammace. I tried adjusting the code to "fix" that, but it didn't seem to help. That's probably still the issue though.

**Comment by cammace** [comment]
fixed in #299, The issue was that we were updating the marker position before the marker views `onViewAdded` was invoked causing it to never be called for some reason. This might be a potential SDK bug @tobrun?
